soapclient/SforceEnterpriseClient.php

```diff
@@ -119,8 +113,9 @@ public function update($sObjects, $type, $assignment_header = NULL, $mru_header
    */
   public function upsert($ext_Id, $sObjects, $type = 'Contact') {
     $arg = new stdClass;
+    $arg->sObjects = [];

```

Comment:
Same remark, you could use the "classic" array syntax for backward compatibility

$arg->sObjects = array();